resources/views/components/organisms/_o-table-of-contents.blade.php

```diff
@@ -1,21 +1,44 @@
 <div class="o-accordion{{ (isset($variation)) ? ' '.$variation : '' }}" data-behavior="accordion">
     @foreach ($items as $item)
+        @php
+            if (isset($currentArticle)) {
+                $isArticleInTree = function ($items) use (&$isArticleInTree, $currentArticle) {
+                    foreach ($items as $childItem) {
+                        if ($childItem->id === $currentArticle->id) {
+                            return true;
+                        }
+                        if (count($childItem->children) > 0 && $isArticleInTree($childItem->children)) {
+                            return true;
+                        }
+                    }
+                    return false;
+                };
```

Comment:
This is a generic function, so assign it to `$isArticleInTree` outside of the `foreach` loop so you're not reassigning the function in each iteration of the loop.

Better yet, make this a function on the model or presenter, so we're not defining business logic inside the view. Call it via `$currentArticle->isInTree($item->children)` or `$currentArticle->present()->isInTree($item->children)`.